web_timeline/static/src/js/web_timeline.js

```diff
@@ -0,0 +1,514 @@
+        return _.isUndefined(value) || _.isNull(value);
+    }
+
+    var TimelineView = View.extend({
+        template: "TimelineView",
+        display_name: _lt('Timeline'),
+        icon: 'fa-clock-o',
+        quick_create_instance: widgets.QuickCreate,
+
+        init: function (parent, dataset, view_id, options) {
+            this._super(parent);
+            this.ready = $.Deferred();
+            this.permissions = {};
+            this.set_default_options(options);
+            this.dataset = dataset;
+            this.model = dataset.model;
+            this.fields_view = {};
+            this.view_id = view_id;
+            this.view_type = 'timeline';
+            this.color_map = {};
+            this.range_start = null;
+            this.range_stop = null;
+            this.selected_filters = [];
+            this.current_window = null;
+        },
+
+        get_perm: function(name){
+            var self = this;
+            var promise = self.permissions[name];
+            if(!promise) {
+                var defer = $.Deferred();
+                new Model(this.dataset.model)
+                .call("check_access_rights", [name, false])
+                .then(function (value) {
+                    self.permissions[name] = value;
+                    defer.resolve();
+                });
+                return defer;
+            } else {
+              return promise;
+            }
+        },
+
+        set_default_options: function(options) {
+            this._super(options);
+            _.defaults(this.options, {
+                confirm_on_delete: true
+            });
+        },
+
+        parse_colors: function(){
+            if(this.fields_view.arch.attrs.colors) {
+                this.colors = _(this.fields_view.arch.attrs.colors.split(';')).chain().compact().map(function(color_pair) {
+                    var pair = color_pair.split(':'), color = pair[0], expr = pair[1];
+                    var temp = py.parse(py.tokenize(expr));
+                    return {'color': color, 'field': temp.expressions[0].value, 'opt': temp.operators[0], 'value': temp.expressions[1].value};
+                }).value();
+            }
+        },
+
+        view_loading: function (fv) {
+            /* xml view timeline options */
+            var attrs = fv.arch.attrs;
+            var self = this;
+            this.fields_view = fv;
+            this.parse_colors();
+            this.$timeline = this.$el.find(".oe_timeline_widget");
+            this.$el.find(".oe_timeline_button_today").click(self.on_today_clicked);
+            this.$el.find(".oe_timeline_button_scale_day").click(self.on_scale_day_clicked);
+            this.$el.find(".oe_timeline_button_scale_week").click(self.on_scale_week_clicked);
+            this.$el.find(".oe_timeline_button_scale_month").click(self.on_scale_month_clicked);
+            this.$el.find(".oe_timeline_button_scale_year").click(self.on_scale_year_clicked);
```

Comment:
This is probably not doing what you expect. It does not matter if you call `on_scale_year_clicked` from `this` or from `self`. The main confusion about the `this` keyword in JS is that, unlike in Python and other languages, JS sets it at calling time, not at definition or bind time. jQuery says that, inside the event execution context, `this = event.target`.

You need a proxy that gets that and applies the context you want, so you should use `$.proxy(this.on_scale_year_clicked, this)` (and same for all the above).

BTW [`self` is already in scope](http://www.w3schools.com/jsref/prop_win_self.asp), so it is not a good name for this kind of stuff (but you don't need it anyway if you use proxies).